src/main/frontend/components/block.cljs

```diff
@@ -802,524 +1171,193 @@
-                                                 (nth arguments 3)
-                                                 (first arguments)
-                                                 (util/safe-parse-int (nth arguments 1))
-                                                 (util/safe-parse-int (nth arguments 2))))
-                                  3
-                                  (when (and (util/safe-parse-int (nth arguments 1))
-                                             (util/safe-parse-int (nth arguments 2)))
-                                    (util/format "[:img {:src \"%s\" :style {:width %s :height %s}}]"
-                                                 (first arguments)
-                                                 (util/safe-parse-int (nth arguments 1))
-                                                 (util/safe-parse-int (nth arguments 2))))
-
-                                  2
-                                  (cond
-                                    (util/safe-parse-int (nth arguments 1))
-                                    (util/format "[:img {:src \"%s\" :style {:width %s}}]"
-                                                 (first arguments)
-                                                 (util/safe-parse-int (nth arguments 1)))
-                                    (contains? #{"left" "right" "center"} (string/lower-case (nth arguments 1)))
-                                    (util/format "[:img.%s {:src \"%s\"}]"
-                                                 (string/lower-case (nth arguments 1))
-                                                 (first arguments))
-                                    :else
-                                    macro-content)
-
-                                  macro-content)
-
-                                (and (seq arguments) macro-content)
-                                (block/macro-subs macro-content arguments)
-
-                                :else
-                                macro-content)
-                macro-content (when macro-content
-                                (template/resolve-dynamic-template! macro-content))]
-            (render-macro config name arguments macro-content format))
-          (let [macro-content (or
-                               (get (state/get-macros) name)
-                               (get (state/get-macros) (keyword name)))
-                format (get-in config [:block :block/format] :markdown)]
-            (render-macro config name arguments macro-content format)))))
+      (get @macro/macros name)
+      ((get @macro/macros name) config options)
 
-    :else
-    ""))
+      :else
+      (macro-else-cp config arguments))))
+
+(defn- emphasis-cp
+  [config kind data]
+  (let [elem (case kind
+               "Bold" :b
+               "Italic" :i
+               "Underline" :ins
+               "Strike_through" :del
+               "Highlight" :mark)]
+    (->elem elem (map-inline config data))))
+
+(defn inline
```

Comment:
Thanks for refactoring this fn. 😄  This will make it easier to make changes and understand this code